tests/test_models/test_inference/test_forecast.py

```diff
@@ -800,3 +836,109 @@ def test_forecast_subset_segments(self, model, transforms, example_tsds):
+        train_ts = select_segments_subset(ts=deepcopy(ts), segments=train_segments)
+        test_ts = select_segments_subset(ts=deepcopy(ts), segments=forecast_segments)
+
+        # fitting
+        train_ts.fit_transform(transforms)
+        test_ts.transform(train_ts.transforms)
+        model.fit(train_ts)
+
+        # forecasting
+        import torch  # TODO: remove after fix at issue-802
+
+        torch.manual_seed(11)
+
+        forecast_ts = test_ts.make_future(future_steps=prediction_size, tail_steps=model.context_size)
+        forecast_ts = make_forecast(model=model, ts=forecast_ts, prediction_size=prediction_size)
+
+        # checking
+        forecast_df = forecast_ts.to_pandas(flatten=True)
+        assert not np.any(forecast_df["target"].isna())
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (CatBoostModelMultiSegment(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (LinearMultiSegmentModel(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (ElasticMultiSegmentModel(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (RNNModel(input_size=1, encoder_length=7, decoder_length=7, trainer_params=dict(max_epochs=1)), []),
+            (
+                MLPModel(input_size=2, hidden_size=[10], decoder_length=7, trainer_params=dict(max_epochs=1)),
+                [LagTransform(in_column="target", lags=[5, 6])],
+            ),
+        ],
+    )
+    def test_forecast_new_segments(self, model, transforms, example_tsds):
+        self._test_forecast_new_segments(example_tsds, model, transforms, train_segments=["segment_1"])
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (
+                DeepARModel(max_epochs=1, learning_rate=[0.01]),
+                [
+                    PytorchForecastingTransform(
+                        max_encoder_length=5,
+                        max_prediction_length=5,
+                        time_varying_known_reals=["time_idx"],
+                        time_varying_unknown_reals=["target"],
+                        target_normalizer=GroupNormalizer(groups=["segment"]),
+                    )
+                ],
+            ),
+            (
+                TFTModel(max_epochs=1, learning_rate=[0.01]),
+                [
+                    PytorchForecastingTransform(
+                        max_encoder_length=21,
+                        min_encoder_length=21,
+                        max_prediction_length=5,
+                        time_varying_known_reals=["time_idx"],
+                        time_varying_unknown_reals=["target"],
+                        static_categoricals=["segment"],
+                        target_normalizer=None,
+                    )
+                ],
+            ),
+        ],
+    )
+    def test_forecast_new_segments_failed_encoding_error(self, model, transforms, example_tsds):
```

Comment:
Are we going to fix them somehow, as I understand we don't?

Reply:
I can mark the as `to_be_fixed`, but I'm not really sure that we will fix this in the near future. As you can see, there are a lot of problems with this models.

Reply:
May be we can create some categories like "backlog"/"todo" for such cases, just to know that sth can be fixed but we don't have time for this